src/graph.cpp

```diff
@@ -61,24 +63,30 @@ void printstocknameandoverall(
     cout << endl;
 }
 
-void printvector(
-    vector<vector<string>> vectorname, vector<string> color, int width, int height) {
-    int colorint;
-    for (int i = 0; i < height; i++) {
-        for (int j = 0; j < width; j++) {
-            colorint = j - 9;
-            if (colorint < 0) {
-                colorint = width - 10;
-            }
-            if (color[colorint] == "green") {
-                cout << "\033[1;31m" << vectorname[j][i] << "\033[0m";
+void printgraphblocks(const vector<vector<string>> & screenElements,
+    const vector<string> & specifiedColorCoordinates, const int width,
+    const int height) {
+    // Boundary check of width and height, otherwise may access screenElements out of
+    // bounds
+    assert(width <= static_cast<int>(screenElements.size()) &&
+           "Width exceeds screenElements width");
+    assert(height <= static_cast<int>(screenElements[0].size()) &&
```

Comment:
Why are you && a sring?

Reply:
Shit

Reply:
Oh I realized it is a common technique to display assert() messages, since assert() don't have two arguments..
The string part always evaluates to `true`, and the assertion primarily depends on the left statement `width <=static_cast<int>(screenElements.size())` now :o (learnt a new thing! xd)
```cpp
#define assert(_Expression) (void) ((!!(_Expression)) || (_wassert(_CRT_WIDE(#_Expression),_CRT_WIDE(__FILE__),__LINE__),0))
```

Reply:
More information: https://stackoverflow.com/a/3692961/14278613